File: 2611-mice-and-cheese/2611-mice-and-cheese.py
class Solution:
    def miceAndCheese(self, r1: List[int], r2: List[int], k: int) -> int:
        d=[]
        for i in range(len(r1)):
            d.append([r1[i]-r2[i],i])
        d.sort(reverse=True)
        res=0
        c=0
        for i in range(k):
            res+=r1[d[i][1]]
            c+=1
        for j in range(c,len(r1)):
            res+=r2[d[j][1]]
        return res
        
        

File: 2611-mice-and-cheese/2611-mice-and-cheese.cpp
class Solution {
public:
    int miceAndCheese(vector<int>& reward1, vector<int>& reward2, int k) {
        vector<pair<int,int>>indexes;
        for(int i=0;i<reward1.size();i++){
            indexes.push_back({reward1[i]-reward2[i],i});
        }
        sort(indexes.begin(),indexes.end());
        reverse(indexes.begin(),indexes.end());
        int r1=0,i;
        for(i=0;i<k;i++){
            auto x=indexes[i];
            int ind=x.second;
            r1+=reward1[ind];
        }
        
        for(;i<reward1.size();i++){
            auto x=indexes[i];
            int ind=x.second;
            r1+=reward2[ind];
        }
        return r1;
    }
    
};